packages/1155-contracts/test/nft/ZoraCreator1155.t.sol

```diff
@@ -1135,6 +1148,118 @@ contract ZoraCreator1155Test is Test {
+        emit RewardsDeposit(
+            collaborator,
+            zora,
+            zora,
+            collaborator,
+            zora,
+            address(target),
+            settings.creatorReward,
+            settings.createReferralReward,
+            settings.mintReferralReward,
+            settings.firstMinterReward,
+            settings.zoraReward
+        );
+        target.mintWithRewards{value: totalReward}(simpleMinter, tokenId, quantity, abi.encode(recipient), address(0));
+
+        assertEq(protocolRewards.balanceOf(collaborator), settings.creatorReward + settings.firstMinterReward);
+    }
+
+    function test_CreatorRewardRecipientConditionalAddress() public {
+        ICreatorRoyaltiesControl.RoyaltyConfiguration memory royaltyConfig;
+        address creatorRewardRecipient;
+
+        address collaborator = makeAddr("collaborator");
+        uint256 quantity = 100;
+
+        init();
+
+        vm.prank(admin);
+        uint256 tokenId = target.setupNewToken("test", quantity);
+
+        (, , address contractFundsRecipient, , , ) = target.config();
+
+        creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+        assertEq(creatorRewardRecipient, contractFundsRecipient);
+
+        royaltyConfig = ICreatorRoyaltiesControl.RoyaltyConfiguration(0, 0, collaborator);
+        vm.prank(admin);
+        target.updateRoyaltiesForToken(tokenId, royaltyConfig);
+
+        creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+        assertEq(creatorRewardRecipient, collaborator);
+
+        royaltyConfig = ICreatorRoyaltiesControl.RoyaltyConfiguration(0, 0, address(0));
+        vm.prank(admin);
+        target.updateRoyaltiesForToken(tokenId, royaltyConfig);
+
+        vm.prank(admin);
+        target.setFundsRecipient(payable(address(0)));
+
+        creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+        assertEq(creatorRewardRecipient, address(target));
+    }
+
+    function test_ContractAsCreatorRewardRecipientFallback() public {
+        uint256 quantity = 100;
+
+        init();
+
+        vm.startPrank(admin);
+        uint256 tokenId = target.setupNewToken("test", quantity);
+
+        target.setFundsRecipient(payable(address(0)));
+
+        target.addPermission(tokenId, address(simpleMinter), adminRole);
+        vm.stopPrank();
+
+        RewardsSettings memory settings = target.computeFreeMintRewards(quantity);
+
+        uint256 totalReward = target.computeTotalReward(quantity);
+        vm.deal(collector, totalReward);
+
+        address creatorRewardRecipient = target.getCreatorRewardRecipient(tokenId);
+
+        vm.prank(collector);
+        target.mintWithRewards{value: totalReward}(simpleMinter, tokenId, quantity, abi.encode(recipient), address(0));
+
+        assertEq(creatorRewardRecipient, address(target));
+        assertEq(protocolRewards.balanceOf(address(target)), settings.creatorReward + settings.firstMinterReward);
```

Comment:
minor, but would be nice to show in this test how that eth can be withdrawn and by who

Reply:
added 👍